docs/guides/benchmarks.md

```diff
@@ -0,0 +1,162 @@
+# Benchmarks
+
+You benchmark your machine learning code in nnbench by encapsulating the benchmarking code in separate functions and applying one of the decorators provided. 
+The available decorators are 
+- `@benchmark`, which runs a benchmark with supplied parameters,
+- `@parametrize`, which runs several benchmarks with the supplied parameter configurations,
+- `@product`, which runs benchmarks with all parameter combinations that arise from the supplied values. 
+
+We will first introduce a small machine learning example which we will subsequently use to showcase the motivation for and usage of the benchmarking decorators.
+
+We recommend to compartmentalize the logic of the machine learning model, the benchmarks, and benchmark execution into different files. In this guide, these are called `training.py`, `benchmarks.py`, and `run_benchmarks.py`.
+
+## Example
+Let us consider an example where we want to evaluate a `scikit-learn`'s `RandomForestClassifier` on the "Iris" example dataset.
+For this purpose, we will define several helper functions inside a file, `training.py`. We use `prepare_data()`, to load the dataset,  `train_rf()` to train a random forest model with the specified parameters, and `accuracy()` to calculate the accuracy of the supplied model on the given dataset.
+
+```python
+# training.py
+import numpy as np
+from sklearn import base, metrics
+from sklearn.datasets import load_iris
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+
+
+def prepare_data() -> tuple[np.ndarray, np.ndarray,np.ndarray, np.ndarray]:
+    data = load_iris()
+    X, y = data.data, data.target
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
+    return X_train, X_test, y_train, y_test
+
+
+def train_rf(X_train: np.ndarray, y_train: np.ndarray, n_estimators: int, max_depth: int, random_state: int = 42) -> RandomForestClassifier:
+    model = RandomForestClassifier(n_estimators=n_estimators, max_depth=max_depth, random_state=random_state)
+    model.fit(X_train, y_train)
+    return model
+
+
+def accuracy(model: base.BaseEstimator, y_test: np.ndarray, y_pred: np.ndarray) -> float:
+    accuracy = metrics.accuracy_score(y_test, y_pred)
+    return accuracy
+```
+
+## `@benchmark` for single benchmarks
+Now, inside a new file called `benchmarks.py` we define our benchmarks.
+We first encapsulate the benchmark logic into a function, `benchmark_accuracy()` which prepares the data, trains a classifier, and lastly, obtains the accuracy.
```

Comment:
I mean, if we're doing this, we might as well make it the whole pipeline. Realistically, model training and preparation should not be a part of the function.